### PR TITLE
XIVY-9347 Use uuid for task instead of id

### PR DESCRIPTION
**Pull request by alexsuter**

### CODE REVIEW

File: dev-workflow-ui/src/ch/ivyteam/workflowui/tasks/TaskModel.java [hunk 1]
```diff
@@ -233,4 +233,12 @@ private static String buildViewerLink(ITask task) {
...
     return ProcessViewer.of(task).isViewAllowed();
   }
+
+  public String getDetailUrl() {
+    return toDetailUrl(uuid);
+  }
+
+  public static String toDetailUrl(String uuid) {
+	return "taskDetails.xhtml?task=" + uuid;
+  }
```
Comment (jackra1n) [inline dev-workflow-ui/src/ch/ivyteam/workflowui/tasks/TaskModel.java:243]: formatting

File: dev-workflow-ui/src/ch/ivyteam/workflowui/util/TaskUtil.java [hunk 2]
```diff
@@ -47,30 +47,30 @@ public static String getPriorityIcon(ITask task) {
     };
   }
 
-  public static ITask getTaskById(long id) {
-    return IWorkflowContext.current().findTask(id);
+  public static ITask getTaskById(String uuid) {
+	return IWorkflowContext.current().findTask(uuid);
   }
```
Comment (jackra1n) [inline dev-workflow-ui/src/ch/ivyteam/workflowui/util/TaskUtil.java:52]: formatting

File: dev-workflow-ui/webContent/includes/components/allTasksTable.xhtml [hunk 3]
```diff
@@ -26,9 +26,9 @@
     </p:column>
 
     <p:column headerText="Name" filterBy="#{task.name}" sortBy="#{task.name}">
-      <h:commandLink action="#{tasksIvyDevWfBean.redirectToTaskDetails(task.id)}">
+      <h:commandLink action="#{tasksIvyDevWfBean.redirectToTaskDetails(task)}">
         <h:outputText id="taskName" value="#{task.name}" />
-        <p:tooltip for="taskName" value="Task ID: #{task.id}" position="top" />
+        <p:tooltip for="taskName" value="Task: #{task.uuid}" position="top" />
```
Comment (jackra1n) [inline dev-workflow-ui/webContent/includes/components/allTasksTable.xhtml:31]: I think I would leave `Task ID: ...` or `Task: UUID: ...`
i feel like if you only say `Task: ...` and show some weird numbers people might think it should be name and something is broken?
just my thoughts

**Review by jackra1n — APPROVED** [review]
other than that very nice 👍🏼

**Comment by jackra1n** [comment]
Also im not sure if you saw it but ES reindexing takes now quite a bit longer 😞 
image

File: dev-workflow-ui/webContent/includes/components/allTasksTable.xhtml (hunk 3 above)
Comment (alexsuter) [reply dev-workflow-ui/webContent/includes/components/allTasksTable.xhtml:31]: I'll discuss it Bruno, when reviewing the story.

**Comment by alexsuter** [comment]
> Also im not sure if you saw it but ES reindexing takes now quite a bit longer disappointed image

related to UUID?

**Comment by jackra1n** [comment]
> > Also im not sure if you saw it but ES reindexing takes now quite a bit longer disappointed
> 
> related to UUID?

im not 110% sure. but it started being slower on 12th of May and i saw you merged UUID on 11th?

**Pull request merged**